File: app/src/main/java/com/zamzzz/killwifi/SettingsActivity.java
package com.zamzzz.killwifi;

import android.os.Bundle;
import android.view.View;
import android.widget.ImageView;
import android.widget.SeekBar;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.switchmaterial.SwitchMaterial;

public class SettingsActivity extends AppCompatActivity {

    private ImageView ivBack;
    private SwitchMaterial swSound, swVibrate, swAnimation, swAutoRetry;
    private SeekBar sbPower, sbThreads;
    private TextView tvPowerValue, tvThreadsValue;
    private MaterialButton btnSave, btnReset;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        initViews();
        setupListeners();
        loadSettings();
    }

    private void initViews() {
        ivBack = findViewById(R.id.iv_back);
        swSound = findViewById(R.id.sw_sound);
        swVibrate = findViewById(R.id.sw_vibrate);
        swAnimation = findViewById(R.id.sw_animation);
        swAutoRetry = findViewById(R.id.sw_auto_retry);
        sbPower = findViewById(R.id.sb_power);
        sbThreads = findViewById(R.id.sb_threads);
        tvPowerValue = findViewById(R.id.tv_power_value);
        tvThreadsValue = findViewById(R.id.tv_threads_value);
        btnSave = findViewById(R.id.btn_save);
        btnReset = findViewById(R.id.btn_reset);
    }

    private void setupListeners() {
        // BACK
        ivBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
                overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
            }
        });

        // POWER SEEK
        sbPower.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                tvPowerValue.setText(progress + "%");
            }
            @Override public void onStartTrackingTouch(SeekBar seekBar) {}
            @Override public void onStopTrackingTouch(SeekBar seekBar) {}
        });

        // THREADS SEEK
        sbThreads.setOnSeekBarChangeListener(new SeekBar.OnSeekBarChangeListener() {
            @Override
            public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
                int threads = (progress + 1) * 100;
                tvThreadsValue.setText(threads + " Threads");
            }
            @Override public void onStartTrackingTouch(SeekBar seekBar) {}
            @Override public void onStopTrackingTouch(SeekBar seekBar) {}
        });

        // SAVE
        btnSave.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(SettingsActivity.this,
                    "✅ Settings Disimpan Tuan!", Toast.LENGTH_SHORT).show();
                finish();
            }
        });

        // RESET
        btnReset.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                sbPower.setProgress(100);
                sbThreads.setProgress(4);
                swSound.setChecked(true);
                swVibrate.setChecked(true);
                swAnimation.setChecked(true);
                swAutoRetry.setChecked(false);
                Toast.makeText(SettingsActivity.this,
                    "🔄 Reset ke Default!", Toast.LENGTH_SHORT).show();
            }
        });
    }

    private void loadSettings() {
        // DEFAULT VALUES
        sbPower.setProgress(100);
        sbThreads.setProgress(4);
        swSound.setChecked(true);
        swVibrate.setChecked(true);
        swAnimation.setChecked(true);
        swAutoRetry.setChecked(false);
    }

    @Override
    public void onBackPressed() {
        finish();
        overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);
    }
}

File: app/src/main/java/com/app/vietsubai/SettingsActivity.kt
package com.app.vietsubai

import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity

class SettingsActivity : AppCompatActivity() {
    private lateinit var store: ApiKeyStore
    private lateinit var geminiInput: EditText
    private lateinit var groqInput: EditText
    private lateinit var status: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        findViewById<android.view.ViewGroup>(android.R.id.content).getChildAt(0).animateEntrance()
        store = ApiKeyStore(this)
        geminiInput = findViewById(R.id.geminiKeyInput); groqInput = findViewById(R.id.groqKeyInput); status = findViewById(R.id.settingsStatus)
        showConfiguredState()
        findViewById<Button>(R.id.saveKeysButton).setOnClickListener { saveKeys() }
        findViewById<Button>(R.id.saveKeysButton).enablePressMotion(); findViewById<Button>(R.id.clearKeysButton).enablePressMotion()
        findViewById<Button>(R.id.clearKeysButton).setOnClickListener { store.clear(); geminiInput.text.clear(); groqInput.text.clear(); showConfiguredState() }
    }

    private fun saveKeys() {
        val gemini = geminiInput.text.toString().trim(); val groq = groqInput.text.toString().trim()
        if (gemini.isBlank() || groq.isBlank()) { status.text = "Cần nhập đủ Gemini API key và Groq API key."; return }
        store.save(gemini, groq); geminiInput.text.clear(); groqInput.text.clear(); status.text = "Đã lưu key mã hóa. Key không được hiển thị lại trên màn hình."
    }
    private fun showConfiguredState() { status.text = if (store.isConfigured()) "Trạng thái: đã cấu hình Gemini và Groq." else "Trạng thái: chưa cấu hình đủ API key." }
}
